tensorflow/lite/micro/docs/rfc-16x8-quantization-port.md

```diff
@@ -0,0 +1,138 @@
+# TensorFlow Lite for Microcontrollers Port of 16x8 Quantized Operators
+
+| Status        | Proposed                                             |
+:-------------- |:---------------------------------------------------- |
+| **RFC #**     | [46767](https://github.com/tensorflow/tensorflow/pull/46767)|
+| **Author(s)** | Daniel Situnayake (me@example.org)                   |
+| **Sponsor**   | Pete Warden (petewarden@google.com)                  |
+| **Updated**   | 2021-01-28                                           |
+
+## Objective
+
+TensorFlow Lite has kernel implementations that support 8 bit quantized weights
+but use 16 bit activations. We wish to port these implementations to TensorFlow
+Lite for Microcontrollers. The increased precision available for activations can
+improve performance for some quantized models.
+
+Arm have agreed to support the initiative by adding the necessary 16x8 APIs to
+CMSIS-NN and porting the CMSIS-NN kernels.
+
+### Goals
+- Port a subset of 16x8 reference kernels from TensorFlow Lite to TensorFlow Lite Micro
+- Avoid increasing default code size of TensorFlow Lite Micro
```

Comment:
Not increasing the binary size (beyond an unavoidable, well-characterized and well-understood amount) is a good goal.

I would add the same for the arena size. This would mostly involve making sure any allocations from the arena needed for 16x8 are only performed for that code-path. To guard against this arena size increase, we will need to add check to the benchmarks that makes use of the [RecordingMemoryAllocator](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/micro/docs/memory_management.md#recording-memory-apis).